lib/Util.cpp

```diff
@@ -37,7 +38,10 @@ string Util::exec(const string cmd) {
 
     tulog.debug("◸", result, "◿");
     if (rc != EXIT_SUCCESS) {
-        throw ExecutionException{"`" + cmd + "` returned with error code " + to_string(rc) + ".", rc, result};
+      if (WIFEXITED(rc))
+          throw ExecutionException{"`" + cmd + "` returned with error code " + to_string(WEXITSTATUS(rc)) + ".", WEXITSTATUS(rc), result};
```

Comment:
This way the catcher of the exception won't be able to distinguish signal exit from plain exit, as the info has been stripped from the return code

```suggestion
          throw ExecutionException{"`" + cmd + "` returned with error code " + to_string(WEXITSTATUS(rc)) + ".", rc, result};
```

Reply:
Ok. I am using -1 to WIFSIGNALED

Reply:
IMO it shouldn't pass the exit code straight through, but have a specific one to indicate plugin failure or just `1` for failure. That's something for @laenion though